### ISSUE TITLE
Abort in reconstruct.cpp:352

### ISSUE THREAD

**Issue by SEU-SSL** [thorfdbg/libjpeg#84]
Here is the command and stack trace:

`./jpeg poc2 /dev/null`

```
(gdb) bt
#0  0x00007ffadbb1f00b in raise () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffadbafe859 in abort () from /lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffadbb6926e in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffadbb712fc in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x00007ffadbb72fac in ?? () from /lib/x86_64-linux-gnu/libc.so.6
#5  0x000000000041bb3d in Reconstruct (infile=0x7ffd85d45942 "poc2", outfile=0x7ffd85d45947 "/dev/null", 
    colortrafo=1, alpha=0x0, upsample=true) at reconstruct.cpp:352
#6  0x0000000000406eaa in main (argc=3, argv=0x7ffd85d43d08) at main.cpp:747
```

[poc2.zip](https://github.com/thorfdbg/libjpeg/files/10789086/poc2.zip)

**Comment by bladchan** [comment]
Hi, SEU-SSL
I used your poc but didn't trigger any bug.
Here's my output:
```
./jpeg ./poc2 /dev/null
jpeg Copyright (C) 2012-2018 Thomas Richter, University of Stuttgart
and Accusoft

For license conditions, see README.license for details.

*** Warning -1038 in Tables::ParseTables, line 1399, file tables.cpp
*** Reason is: found invalid marker, probably a marker size is out of range

*** Warning -1038 in Frame::StartParseHiddenScan, line 875, file frame.cpp
*** Reason is: Start of Scan SOS marker missing

*** Warning -1038 in Frame::ParseTrailer, line 1091, file frame.cpp
*** Reason is: missing an EOI marker at the end of the stream

*** Warning -1038 in Image::ParseTrailer, line 1468, file image.cpp
*** Reason is: expecting an EOI marker at the end of the stream


0 bytes memory not yet released.

15337 bytes maximal required.

84 allocations performed.
```
BTW, I used clang 6.0 to compile this library. Maybe this error is related to compiler environment?

**Comment by SEU-SSL** [comment]
> Hi, SEU-SSL I used your poc but didn't trigger any bug. Here's my output:
> 
> ```
> ./jpeg ./poc2 /dev/null
> jpeg Copyright (C) 2012-2018 Thomas Richter, University of Stuttgart
> and Accusoft
> 
> For license conditions, see README.license for details.
> 
> *** Warning -1038 in Tables::ParseTables, line 1399, file tables.cpp
> *** Reason is: found invalid marker, probably a marker size is out of range
> 
> *** Warning -1038 in Frame::StartParseHiddenScan, line 875, file frame.cpp
> *** Reason is: Start of Scan SOS marker missing
> 
> *** Warning -1038 in Frame::ParseTrailer, line 1091, file frame.cpp
> *** Reason is: missing an EOI marker at the end of the stream
> 
> *** Warning -1038 in Image::ParseTrailer, line 1468, file image.cpp
> *** Reason is: expecting an EOI marker at the end of the stream
> 
> 
> 0 bytes memory not yet released.
> 
> 15337 bytes maximal required.
> 
> 84 allocations performed.
> ```
> 
> BTW, I used clang 6.0 to compile this library. Maybe this error is related to compiler environment?

Hello, I guess this issue is fixed by this commit [db33a6e](https://github.com/thorfdbg/libjpeg/commit/db33a6ee2d034eea2c62fd3c5dcb6fa86746579f)
Perhaps you can try to reproduce the vulnerability with an earlier version.
